src/waveform/overviews/waveformoverviewrenderer.h

```diff
@@ -0,0 +1,60 @@
+#pragma once
+
+#include <QColor>
+
+#include "util/singleton.h"
+#include "waveform/waveform.h"
+#include "widget/woverview.h"
+
+class QPainter;
+
+class WaveformOverviewRenderer : public Singleton<WaveformOverviewRenderer> {
+  public:
+    QImage render(ConstWaveformPointer, WOverview::Type type);
+    void drawWaveformPartRGB(
```

Comment:
this doesn't have internal state, nor does it need to be a singleton. Consider making this free functions instead and removing the static data. Pass it instead when calling the functions.